conda/common/logic.py

```diff
@@ -396,9 +623,28 @@ def LinearBound_(self, equation, lo, hi, preprocess, polarity):
         return res
 
     def LinearBound(self, equation, lo, hi, preprocess=True, polarity=None, name=None):
-        return self.Eval_(self.LinearBound_, (equation, lo, hi, preprocess),
-                          polarity, name, conv=False)
+        if log.isEnabledFor(DEBUG):
+            old_clause_count = self.get_clause_count()
+        ret = self.Eval_(
+            self.LinearBound_, (equation, lo, hi, preprocess), polarity, name, conv=False)
+        if log.isEnabledFor(DEBUG):
+            new_clause_count = self.get_clause_count() - old_clause_count
+            log.debug(
+                'new clauses: %d (%.0f%%)',
+                new_clause_count, 100.0 * new_clause_count / max(1, old_clause_count),
+            )
+        return ret
+
+    @time_recorder(module_name=__name__)
+    def _run_sat(self, clauses, m, limit=0):
+        if log.isEnabledFor(DEBUG):
+            log.debug("Invoking SAT with clause count: %s", self.get_clause_count())
+        solution = PycoSatSolver().run(clauses, m, limit=limit)
+        # solution = PySatSolver().run(clauses, m)
+        # solution = CryptoMiniSatSolver().run(clauses, m, threads=1)
```

Comment:
1. `conda install conda-forge::cryptominisat`
2. use this line instead of the `PycoSatSolver` one
3. solve problems we were not able to save before
4. success

No, seriously, if we can fix the slow setup time for CryptoMiniSat upstream, we should definitely use that solver, at least given my initial testing.
With that solver I could even get a solution for
`conda create -c bioconda -c conda-forge -c defaults qualimap=2.2.2a=3`
which is my go-to for screwing the solver up. With PicoSAT I let this one once run for, I think, 30 **hours** or so and didn't get any solution. With CryptoMiniSat it still took a long time, but I think it was less than 20 minutes or so.

Reply:
Also, no idea about all those variations, but CryptoMiniSat did pretty well in http://sat2018.forsyte.tuwien.ac.at/index.php?cat=results

Reply:
xref: https://github.com/msoos/cryptominisat/issues/513

Reply:
:) Yes, the solver performance is significantly better than PicoSat -- that's an old solver. However, I agree that the setup time can be more than ideal. I'm happy to pour effort into fixing that, though. It's a lot easier to fix than bringing PicoSat up to speed. CMS is highly optimised in many ways, but startup can still be optimised, I agree. I'll comment on the issue opened on CMS. Basically, I'll need a way to reproduce what you have and then I can iterate on it to fix all the bugs :) Very much looking forward to this, as it could fix a longstanding issue in CMS for a good cause.